fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1341,9 +1341,11 @@ private void handleQueryStmt() throws Exception {
 
         // Sql and PartitionCache
         CacheAnalyzer cacheAnalyzer = new CacheAnalyzer(context, parsedStmt, planner);
-        if (cacheAnalyzer.enableCache() && !isOutfileQuery && queryStmt instanceof SelectStmt) {
-            handleCacheStmt(cacheAnalyzer, channel, (SelectStmt) queryStmt);
-            return;
+        if ((!isOutfileQuery && queryStmt instanceof SelectStmt) || queryStmt instanceof LogicalPlanAdapter) {
```

Comment:
nereids also support outfile. if cache do not support outfile, u should check `!queryStmt.logicalPlan instance of LogicalFileSink`

Reply:
fix, thxs